### PR TITLE
Minor cleanups in FS

### PR DESCRIPTION
**Pull request by QuzarDC**
EDIT: Refreshed these changes from spring for the current situation. Collection of mostly formal changes.

- `fs_map_hnd` already sets the errno on its bad paths, so avoid duplicating that.
- Fix typo in dirent documentation.
- Import defines from genromfs and romdisk documentation to fs_romdisk. Demagic volumename/filename maxes.
- Add romdisk write and unlink stubs. Not sure that they're helpful at all, but they enforce more accurate behavior based on the POSIX definitions for what the errnos should be set to.
- Cleanups around shutdowns for some FSes and conversion of some fh values to bools.

### CODE REVIEW

File: kernel/fs/fs_socket.c [hunk 1]
```diff
@@ -298,8 +305,8 @@ int socket(int domain, int type, int protocol) {
         }
     }
 
-    /* If i is NULL, we got through the whole list without finding anything. */
-    if(!i) {
+     /* If i is NULL, we got through the whole list without finding anything. */
+    if(i != NULL) {
```
Comment (Firnis) [inline kernel/fs/fs_socket.c:309]: Sholdn't it be `i == NULL`?

Comment (QuzarDC) [reply kernel/fs/fs_socket.c:277]: Absolutely! Thank you.

**Comment by andressbarajas** [comment]
Why the code style change from if(!h) => if(h == NULL) ?

File: kernel/fs/fs.c [hunk 2]
```diff
@@ -745,10 +739,7 @@ static int fs_vfcntl(file_t fd, int cmd, va_list ap) {
     fs_hnd_t *h = fs_map_hnd(fd);
     int rv;
 
-    if(!h) {
-        errno = EBADF;
```
Comment (gyrovorbis) [inline kernel/fs/fs.c:749]: In case you didn't see it in Discord:

Why was this removed? Was it not how POSIX handles it? Screwing up the upstream?

**Comment by QuzarDC** [comment]
> Why the code style change from if(!h) => if(h == NULL)? Just curious what is the coding style we should use for future PRs

Whoops. I'll find and fix that. Unintentional. For style I'd say only if it's really specifically important to demonstrate `NULL` vs other values, but that'd be rare.

**Review by gyrovorbis — APPROVED** [review]
Awesome! Looks good!
